src/sage/modular/abvar/homspace.py

```diff
@@ -808,12 +806,11 @@ def __init__(self, A, gens=None, category=None):
         self._A = A
 
         # Initialise self with the correct category.
-        # We need to initialise it as a ring first
         if category is None:
             homset_cat = A.category()
         else:
             homset_cat = category
-        # Remark: Ring.__init__ will automatically form the join
+        # Remark: Parent.__init__ will automatically form the join
         # of the category of rings and of homset_cat
```

Comment:
does it? or is that done by `Endsets()`?

Reply:
Well, that's the line `Parent.__init __` that does the job. Once done, the categories contains that of "rings". The next line does not change anything to the list of categories.